BrowserKit/Sources/Redux/Store.swift

```diff
@@ -61,6 +62,12 @@ public class Store<State: StateType>: DefaultDispatchStore {
     }
 
     public func dispatch(_ action: Action) {
+        guard Thread.isMainThread && !actionRunning else {
+            DispatchQueue.main.async { [weak self] in self?.dispatch(action) }
+            return
+        }
+        actionRunning = true; defer { actionRunning = false }
+
```

Comment:
@OrlaM @yoanarios Any thoughts? Temporary quality-of-life improvement until we explore MainActor. It avoids the internal queue in https://github.com/mozilla-mobile/firefox-ios/pull/16777 but just provides an auto-dispatch when necessary. If any concerns though I'm happy to revert and/or revisit later.